### ISSUE TITLE
Samsung ML-1640 doesn't work of Fedora 35

### ISSUE THREAD

**Issue by alex27riva** [OpenPrinting/cups#302]
Hi everyone, I have a problem with Fedora 35 and my laser B/N printer Samsung ML-1640.
The printer is recognized correctly by Fedora, but when I print something the printer heats up, the green led on the printer blinks (like normal) but doesn’t print anything…
The same printer works fine under Windows 10 or Pop_OS 20.04 LTS, so the issue is on Fedora 35 (tested on different PCs).
I attach some logs:
[cups_job25.txt](https://github.com/OpenPrinting/cups/files/7661297/cups_job25.txt)

**Comment by zdohnal** [comment]
Hi @alex27riva ,

thank you for reporting the issue!

From the logs I see:

```
dic 06 14:43:04 fedora cupsd[3210]: [Job 25] Gutenprint: === BEGIN GUTENPRINT SETTINGS ===
dic 06 14:43:04 fedora cupsd[3210]: [Job 25] Gutenprint:     Driver: samsung-ml-1650
```

There is a ML-1640 driver in splix package, have you tried it? Then you can install the printer with model uri ```drv:///splix/splix-samsung.drv/ml1640.ppd```

```
dic 06 14:43:04 fedora cupsd[3210]: [Job 25] Waiting for read thread to exit...
dic 06 14:43:11 fedora cupsd[3210]: [Job 25] Read thread still active, aborting the pending read...
dic 06 14:43:13 fedora cupsd[3210]: [Job 25] Resetting printer.
```

This happens if reading from backchannel can't be ended - but this happens after the file was sent to the printer, so IMO it shouldn't influence the printing itself - it is a problem with cleanup after printing.

To sum it up, it would be great if you tried splix driver instead of Gutenprint driver for a different model and told us if it helps.

**Comment by alex27riva** [comment]
Hi, I installed the splix package and selected the driver from GNOME settings, and it works!
![Schermata del 2021-12-08 10-11-54](https://user-images.githubusercontent.com/5512235/145183307-eff9dcfb-bc44-4cb4-956a-f630db171bf2.png)

**Comment by zdohnal** [comment]
Thanks for letting me know, closing.
